### PR TITLE
Push Image to Private ECR repository

### PR DESCRIPTION
**Pull request by vasireddy99**
**Description:**

This PR allows to push the ADOT Collector image to the private registry during the CI workflow. ADOT operator image duirng the CD workflow

By pushing the image to the private repository it allows us to utilize the ECR enhanced scanning on these images to find vulnerabilities.

**Documentation:** 

N/A

Testing :
 
[local fork](https://github.com/vasireddy99/aws-otel-collector/runs/5892590795?check_suite_focus=true)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -545,6 +547,8 @@ jobs:
           tags: |
             public.ecr.aws/${{ env.ECR_REPO }}:${{ needs.e2etest-preparation.outputs.version }}
             public.ecr.aws/${{ env.ECR_REPO }}:latest
+            611364707713.dkr.ecr.us-east-1.amazonaws.com/${{ env.ECR_REPO }}:${{ needs.e2etest-preparation.outputs.version }}
+            611364707713.dkr.ecr.us-east-1.amazonaws.com/${{ env.ECR_REPO }}:latest
```
Comment (bryan-aguilar) [inline .github/workflows/CI.yml:551]: we do not need a latest tag for this repository

**Comment by bryan-aguilar** [comment]
I do not think these should be pushed during the `e2e-release` step. This would push images that do not pass integration testing. I think a better decision would be to push images to the private repository after CI tests are successful.

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -1191,6 +1191,21 @@ jobs:
...
+        uses: docker/login-action@v1
+        with:
+          registry: {{ secrets.INTEG_TEST_ACCOUNT }}.dkr.ecr.us-east-1.amazonaws.com
+          username: ${{ secrets.INTEG_TEST_AWS_KEY_ID }}
+          password: ${{ secrets.INTEG_TEST_AWS_KEY_SECRET }}
+        env:
+          AWS_REGION: us-east-1
+
+      - name: Pull image from public ECR and upload to Private ECR Repository
+        uses: akhilerm/tag-push-action@v2.0.0
```
Comment (anuraaga) [inline .github/workflows/CI.yml:1204]: Use crane to copy images to not lose multiarch

https://github.com/open-telemetry/opentelemetry-java/blob/main/.github/workflows/docker-test-containers-daily.yml#L33

Comment (bryan-aguilar) [reply .github/workflows/CI.yml:1204]: `tag-push-action` uses crane. See doc for [action](https://github.com/akhilerm/tag-push-action) here.

Comment (bryan-aguilar) [reply .github/workflows/CI.yml:1204]: Although moving away from using vended GitHub actions where we can wouldn't be a bad idea.

File: .github/workflows/CD-adot-operator.yml [hunk 3]
```diff
@@ -40,10 +42,32 @@ jobs:
...
+          VERSION="${{ github.event.inputs.version }}" bash tools/release/adot-operator-images-mirror/version-check.sh
+          echo "::set-output name=version::$VERSION"
 
       - name: Mirror ADOT operator
         if: ${{ steps.version.outputs.update-operator == 'true' }}
         run: cd tools/release/adot-operator-images-mirror && go run ./
         env:
           AWS_SDK_LOAD_CONFIG: true
+
+  push-to-private-ecr:
```
Comment (erichsueh3) [inline .github/workflows/CD-adot-operator.yml:55]: We should probably rework the mirroring tool to push to private ECR first and then public then change this to reflect that

**Review by erichsueh3 — COMMENTED** [review]
Do we need to push the Collector image still?

File: .github/workflows/CD-adot-operator.yml (hunk 3 above)
Comment (vasireddy99) [reply .github/workflows/CD-adot-operator.yml:55]: Sure, I understood your point. 
Scenario  - if there is a decision to not ship the operator image  while the upstream `operator image` has vulnerability, in that case, that functionality needs to separated from CD.

**Comment by vasireddy99** [comment]
> Do we need to push the Collector image still? We can use [this image](https://gallery.ecr.aws/aws-otel-test/adot-collector-integration-test) instead

Initial thought was to scan the `published image`, but it leads to a  discussion whether we scan both of them.

anyhow, we are not going to use this private image for our task.

**Pull request closed without merging**